From 23a9fdf6d80c0b3fcab7a3e307843a14f165ead2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20=C3=85dahl?= <jadahl@gmail.com>
Date: Thu, 5 Nov 2020 07:46:40 +0000
Subject: [PATCH] x11/output: Fix init tile info guard

Monitor tile info is possible to fetch when RANDR version 15 is exposed
by the X11 server. We had inverted the check meaning that only if older
versions were advertised would we attempt to init the tile information.
Fix this guard, thus fix monitor tiling on X11.

Closes: https://gitlab.gnome.org/GNOME/mutter/-/issues/1524


(cherry picked from commit 4ecc80fd8025842452262001fc4d6b7047696d4e)
---
 src/backends/x11/meta-output-xrandr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/backends/x11/meta-output-xrandr.c b/src/backends/x11/meta-output-xrandr.c
index 3a85cb7aa4..b7d18a0d3f 100644
--- a/src/backends/x11/meta-output-xrandr.c
+++ b/src/backends/x11/meta-output-xrandr.c
@@ -848,7 +848,7 @@ meta_output_xrandr_new (MetaGpuXrandr *gpu_xrandr,
       output_info->height_mm = xrandr_output->mm_height;
     }
 
-  if (!meta_monitor_manager_xrandr_has_randr15 (monitor_manager_xrandr))
+  if (meta_monitor_manager_xrandr_has_randr15 (monitor_manager_xrandr))
     output_info_init_tile_info (output_info, xdisplay, output_id);
   output_info_init_modes (output_info, gpu, xrandr_output);
   output_info_init_crtcs (output_info, gpu, xrandr_output);
-- 
GitLab

